### DISCUSSION TITLE
Error 404 Page not found

### DISCUSSION THREAD

**Discussion by shivansh193** [vercel/community#3167] [Help]
### Summary

I created a website for my organization, which was running smoothly on the dev environment but upon hosting, I discovered my inability to access the pages other than the home page, which were '/favorites' '/pets' '/reservations' '/trips', I am not able to access it as I was able to in my dev environment. The project was created on Next JS app dir, I am attaching the file structure for reference.
![image](https://github.com/vercel/community/assets/96665059/5a69ffc0-bdf6-49c2-a2c9-18e504c21dd6)

I am also attaching the middleware.ts file code for reference 
```
export { default } from "next-auth/middleware"

export const config={
    matcher:[
        "/trips",
        "/favorites",
        "/reservations",
        "/pets"

    ]
}
```
While trying npm run build in a local enviorment, I got the following error.
![image](https://github.com/vercel/community/assets/96665059/5eca4c23-2df8-473a-9b7c-e7313043ac9c)

### Example

petner-plat.vercel.app

### Steps to Reproduce

Upon opening the app as mentioned in the example section, one can log in with random account details such as admin123@gmail.com, and try accessing the /favorites route to reproduce the error

**Comment by amyegan** [comment]
Hi @shivansh193. This may be related to your auth configuration for the restricted pages. You can use the [Runtime Logs](https://vercel.com/docs/concepts/observability/runtime-logs#) and [try/catch blocks](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) to get more information about what is happening on the routes that redirect to the 404 page.

**Reply by shivansh193** [reply]
Could you mention, where exactly I should be use try/catch blocks? I did not get any substantial info from the Run time logs, apart from a API call which results in the 404 error upon opening.

**Reply by amyegan** [reply]
You might wrap the contents of a page's exported function or any serverless function code used. The right spots depend on your code structure and project setup.

It could be a problem with the middleware. I see the matcher, but I don't know what the function is actually doing, so I can't say for sure.
